src/cpp/session/modules/rmarkdown/NotebookAlternateEngines.cpp

```diff
@@ -349,6 +349,17 @@ Error executeSqlEngineChunk(const std::string& docId,
    // ensure we always emit an execution complete event on exit
    ChunkExecCompletedScope execScope(docId, chunkId);
 
+   FilePath parentPath = notebook::chunkOutputPath(
+       docId, chunkId, nbCtxId, ContextSaved);
+   error = parentPath.ensureDirectory();
+   if (error)
+   {
+      std::string message = "Failed to create SQL chunk directory";
+      reportChunkExecutionError(docId, chunkId, nbCtxId, message, parentPath);
+        
+      return Success();
```

Comment:
For fault tolerance it might be preferable to go ahead and execute the chunk (since we can still run it and deliver the results to the client), and log a warning that includes the path we weren't able to create.